src/rpm/package.rs

```diff
@@ -216,31 +208,59 @@ impl RPMPackage {
         );
 
         verifier.verify(header_bytes.as_slice(), signature_header_only)?;
+        self.verify_digests()?;
 
         let header_and_content_cursor =
             SeqCursor::new(&[header_bytes.as_slice(), self.content.as_slice()]);
 
         verifier.verify(header_and_content_cursor, signature_header_and_content)?;
 
-        Ok(SignatureVerificationOutcome::Pass)
+        Ok(())
     }
 
-    pub fn verify_digest(&self) -> Result<DigestVerificationOutcome, RPMError> {
+    pub fn verify_digests(&self) -> Result<(), RPMError> {
         let mut header = Vec::<u8>::with_capacity(1024);
         // make sure to not hash any previous signatures in the header
         self.metadata.header.write(&mut header)?;
 
-        let recreated_from_content = Self::create_digests_from_readers(
+        let recreated_from_content = Self::create_legacy_header_digests(
             &mut header.as_slice(),
             SeqCursor::new(&[header.as_slice(), self.content.as_slice()]),
         )?;
 
-        let package_contained = self.metadata.get_digests()?;
-        if recreated_from_content == package_contained {
-            Ok(DigestVerificationOutcome::Match)
-        } else {
-            Ok(DigestVerificationOutcome::Mismatch)
+        let sig_header_digests = self.metadata.get_digests()?;
+
+        let payload_digest_val = self
+            .metadata
+            .header
+            .get_entry_data_as_string_array(IndexTag::RPMTAG_PAYLOADDIGEST);
+        let payload_digest_algo = self
+            .metadata
+            .header
+            .get_entry_data_as_u32(IndexTag::RPMTAG_PAYLOADDIGESTALGO);
+
+        if let (Ok(payload_digest_val), Ok(payload_digest_algo)) =
+            (payload_digest_val, payload_digest_algo)
+        {
+            let mut hasher = match payload_digest_algo {
+                a if a == FileDigestAlgorithm::Sha2_256 as u32 => sha2::Sha256::default(),
+                _ => unimplemented!("An unknown payload digest algorithm was used"),
+                // At the present moment even rpmbuild only supports sha256
+            };
+            let payload_digest = {
+                hasher.update(self.content.as_slice());
+                hex::encode(hasher.finalize())
+            };
+            if payload_digest != payload_digest_val[0] {
+                return Err(RPMError::DigestMismatchError);
+            }
         }
+
+        if recreated_from_content != sig_header_digests {
+            return Err(RPMError::DigestMismatchError);
+        }
+
+        Ok(())
```

Comment:
With the lack of a response type, we don't know if an actual verification took place or if the items were missing. Depending on the context either might be acceptable or not.

Reply:
Added an @todo item